### ISSUE TITLE
Using similarity threshold with PGVector

### ISSUE THREAD

**Issue by uogbuji** [OoriData/OgbujiPT#59]
A vector search will typically return a variety of results with different similarity scores. Right now in `PGVectorHelper.query` we just sort them all and return the one with the highest score. You can change the `limit` arg. We hard-code cosine similarity from the [options PGVector provides](https://github.com/pgvector/pgvector#querying).

Overall I'd like us to think about how we can expose many more of the options through our interface, but for now a glaring omission is the ability to limit responses by a threshold similarity/distance. If there are no good matches, for example, it's more useful to get an empty result set than one or more terrible matches. We want to be able to say "return up to the top N (or all) where similarity score is over S".

cc @chimezie @choccccy

**Comment by chimezie** [comment]
I agree, 100% (re: sorting by similarity threshold).  It is what I tend to end up doing w/ related similarity measures such as Levenstein distance, for e.g.

**Comment by uogbuji** [comment]
I noticed we're returning cosine sim in an opposite way to what's intuitive, where lower scores mean a better match. That make sense when you think of it as distance, but we're using the term "similarity". Also, that means the semantics for PGVector are the opposite to those in Qdrant & Chroma. I've reversed this logic and the tests still pass (as I'd expected).

**Comment by uogbuji** [comment]
`pgvector.py` is way too unwieldy. I'm going to look to split it into a couple of files, with a deprecation path, of course. Also going to look into separating snippets-type DocDBs, as useful for [SKA](https://github.com/OoriData/OgbujiPT/discussions/46), versus proper docs, e.g. traditional RAG.
